MapboxNavigation/NavigationViewController.swift

```diff
@@ -368,10 +369,11 @@ public class NavigationViewController: NavigationPulleyViewController, RouteMapV
             }
         }
         
-        if destination == nil {
-            let annotation = MGLPointAnnotation()
-            annotation.coordinate = route.coordinates!.last!
-            destination = annotation
+        for leg in route.legs {
+            if let last = leg.steps.last {
+                let annotation = MGLPointAnnotation()
+                annotation.coordinate = last.coordinates!.last!
```

Comment:
The step’s `destination` is a Waypoint whose `coordinates` property corresponds to the location we’d want to annotate.